addons/binding/org.openhab.binding.networkcamera/src/main/java/org/openhab/binding/networkcamera/internal/ftp/FtpServer.java

```diff
@@ -0,0 +1,195 @@
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.ftpserver.FtpServerFactory;
+import org.apache.ftpserver.ftplet.DefaultFtplet;
+import org.apache.ftpserver.ftplet.FileSystemFactory;
+import org.apache.ftpserver.ftplet.FileSystemView;
+import org.apache.ftpserver.ftplet.FtpException;
+import org.apache.ftpserver.ftplet.FtpRequest;
+import org.apache.ftpserver.ftplet.FtpSession;
+import org.apache.ftpserver.ftplet.FtpStatistics;
+import org.apache.ftpserver.ftplet.Ftplet;
+import org.apache.ftpserver.ftplet.FtpletContext;
+import org.apache.ftpserver.ftplet.FtpletResult;
+import org.apache.ftpserver.ftplet.User;
+import org.apache.ftpserver.listener.Listener;
+import org.apache.ftpserver.listener.ListenerFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Simple FTP server implementation to receive images from network camera.
+ *
+ *
+ * @author Pauli Anttila - Initial contribution
+ */
+public class FtpServer {
+
+    private Logger logger = LoggerFactory.getLogger(FtpServer.class);
+
+    private int port;
+    int idleTimeout;
+
+    private org.apache.ftpserver.FtpServer server;
+    private static List<FtpServerEventListener> listeners;
+    private MyFTPLet myFTPLet;
+    private FTPUserManager FTPUserManager;
+
+    public FtpServer() {
+        listeners = new ArrayList<FtpServerEventListener>();
+        FTPUserManager = new FTPUserManager();
+    }
+
+    public void startServer(int port, int idleTimeout) {
+        stopServer();
+        this.port = port;
+        this.idleTimeout = idleTimeout;
+        FTPUserManager.setIdleTimeout(idleTimeout);
+        initServer();
+    }
+
+    public void stopServer() {
+        if (server != null) {
+            logger.info("Stopping FTP server");
+            server.stop();
+        }
+    }
+
+    public synchronized void addEventListener(FtpServerEventListener listener) {
+        if (!listeners.contains(listener)) {
+            listeners.add(listener);
+        }
+    }
+
+    public synchronized void addAuthenticationCredentials(String username, String password)
+            throws IllegalArgumentException {
+
+        FTPUserManager.addAuthenticationCredentials(username, password);
+    }
+
+    public synchronized void removeAuthenticationCredentials(String username) {
+        FTPUserManager.removeAuthenticationCredentials(username);
+    }
+
+    public synchronized void removeEventListener(FtpServerEventListener listener) {
+        listeners.remove(listener);
+    }
+
+    private void sendMsgToListeners(String userName, byte[] data) {
+        try {
+            Iterator<FtpServerEventListener> iterator = listeners.iterator();
+
+            while (iterator.hasNext()) {
+                iterator.next().fileReceived(userName, data);
+            }
+
+        } catch (Exception e) {
```

Comment:
Please do not catch `Exception`, catch the specific ones.

Reply:
Callback could throw any kind of execution.